lib/propshaft/asset.rb

```diff
@@ -21,7 +24,7 @@ def length
   end
 
   def digest
-    @digest ||= Digest::SHA1.hexdigest("#{content}#{version}")
+    @digest ||= digest_class.hexdigest("#{content}#{version}")
```

Comment:
Do you think that this would be more flexible/clear if `digest_class` was a `Proc`? I know that the sprockets config was `digest_class`, but if we wanted to provide custom behavior, we'd have to dig in to the source to find out that we call `#hexdigest` on it, but a proc might have a more obvious interface?

```suggestion
    @digest ||= digester.call("#{content}#{version}")
```

and something like in the default configs
```ruby
digester: ->(content_version) { Digest::SHA1.hexdigest(content_version) }
```

Reply:
I think if this were a greenfield feature, I'd agree and prefer something like what you've proposed.

In this case, I definitely leaned more on the Sprockets compatibility angle, even if it ends up being more clunky to use in our specific use-case.

I'd be curious for the maintainer's take on whether Sprockets compatibility is a desirable approach, vs. rethinking how best this option might be exposed.